From f94e7695d914f237738eb004b438536334a3c9b1 Mon Sep 17 00:00:00 2001
From: Langsdorff <131063968+clangsdorff@users.noreply.github.com>
Date: Sun, 4 Oct 2026 23:14:31 +0300
Subject: [PATCH 36/52] sepolicy: let cameraserver read the camera toggles

label the samsung remosaic, samsung 60fps video and grapheneos camera
60fps toggles as phh_samsung_camera_prop, let the treble app set them
and let cameraserver read them.
---
 sepolicy/property_contexts | 3 +++
 sepolicy/samsung.te        | 5 +++++
 2 files changed, 8 insertions(+)

diff --git a/sepolicy/property_contexts b/sepolicy/property_contexts
index 6caee05..90d9239 100644
--- a/sepolicy/property_contexts
+++ b/sepolicy/property_contexts
@@ -1,3 +1,6 @@
 persist.sys.sf.hs_mode         u:object_r:sf_hs_mode_prop:s0
 hwc.exynos.vsync_mode      u:object_r:vendor_hwc_prop:s0
 sys.phh.oplus.fppress    u:object_r:phh_oplus_prop:s0
+persist.sys.phh.samsung.remosaic u:object_r:phh_samsung_camera_prop:s0
+persist.sys.phh.samsung.video60 u:object_r:phh_samsung_camera_prop:s0
+persist.sys.phh.samsung.video60_grapheneos u:object_r:phh_samsung_camera_prop:s0
diff --git a/sepolicy/samsung.te b/sepolicy/samsung.te
index 9446fad..ce4eb7b 100644
--- a/sepolicy/samsung.te
+++ b/sepolicy/samsung.te
@@ -47,3 +47,8 @@ allow kernel tmpfs:file relabelfrom;

 type sysfs_ski_display_writable, file_type;
 allow platform_app sysfs_ski_display_writable:file r_file_perms;
+
+type phh_samsung_camera_prop, property_type;
+set_prop(system_app, phh_samsung_camera_prop);
+set_prop(platform_app, phh_samsung_camera_prop);
+get_prop(cameraserver, phh_samsung_camera_prop);
-- 
2.56.0

